src/internal/m365/graph/concurrency_middleware.go

```diff
@@ -20,34 +20,41 @@ import (
 // "how many calls at one time"
 // ---------------------------------------------------------------------------
 
-// concurrencyLimiter middleware limits the number of concurrent requests to graph API
-type concurrencyLimiter struct {
+// concurrencyLimiterMiddleware middleware limits the number of concurrent requests to graph API
+type concurrencyLimiterMiddleware struct {
 	semaphore chan struct{}
 }
 
 var (
-	once                  sync.Once
-	concurrencyLim        *concurrencyLimiter
-	maxConcurrentRequests = 4
+	once                                sync.Once
+	concurrencyLimitMiddlewareSingleton = &concurrencyLimiterMiddleware{}
+	maxConcurrentRequests               = 4
 )
 
-func generateConcurrencyLimiter(capacity int) *concurrencyLimiter {
+func generateConcurrencyLimiter(capacity int) *concurrencyLimiterMiddleware {
 	if capacity < 1 || capacity > maxConcurrentRequests {
 		capacity = maxConcurrentRequests
 	}
 
-	return &concurrencyLimiter{
+	return &concurrencyLimiterMiddleware{
 		semaphore: make(chan struct{}, capacity),
 	}
 }
 
 func InitializeConcurrencyLimiter(capacity int) {
 	once.Do(func() {
-		concurrencyLim = generateConcurrencyLimiter(capacity)
+		concurrencyLimitMiddlewareSingleton.semaphore = generateConcurrencyLimiter(capacity).semaphore
```

Comment:
Can we add a `turning on concurrency limiter` log here?

Reply:
done